src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -38,10 +42,10 @@ public void OnBeginRequest(HttpContext httpContext, long timestamp)
         {
             if (this.client.IsEnabled())
             {
-                httpContext.Features.Set(new RequestTelemetry());
-
-                this.beginRequestTimestamp.Value = timestamp;
-                this.client.Context.Operation.Id = httpContext.TraceIdentifier;
+                var requestTelemetry = new RequestTelemetry();
+                requestTelemetry.Context.Operation.Id = httpContext.TraceIdentifier;
```

Comment:
Answering my own question, it looks like the original was incorrect. We have an ```OperationIdTelemetryInitializer``` which will supply an OperationId. So, #340 is fixed (I'll also add a test for it).